### PR TITLE
Pushed some HackerRank problems

### PR DESCRIPTION
**Pull request by mathias-onu**

### CODE REVIEW

File: src/hackerrank-problems/breaking-the-records.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        } else if (scores[i] < minRecord) {
+            records[1]++
+            minRecord = scores[i]
+        }
+    }
+    
+    return records
+}
+
+console.log(breakingRecords([12, 24, 10, 24]))
```
Comment (pancudaniel7) [inline src/hackerrank-problems/breaking-the-records.ts]: Looks good, also if you want to use syntax features you can rewrite this code as:

````

function breakRecords(scores) {
  let [max, min] = [scores[0], scores[0]], [maxB, minB] = [0, 0];

  scores.forEach(score => {
    if (score > max) [max, maxB] = [score, maxB + 1];
    else if (score < min) [min, minB] = [score, minB + 1];
  });

  return [maxB, minB];
}

````

File: src/hackerrank-problems/camel-case4.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        if (input.startsWith('S')) {
+            console.log(splitCamelCase(input))
+        }
+        // Combine
+        else {
+            console.log(combineCamelCase(input))
+        }
+        currentLine++
+    }
+}
```
Comment (pancudaniel7) [inline src/hackerrank-problems/camel-case4.ts]: This types of comments are not adding value here, I recommend to use comments just on pure utility functions. And not in code bit above function signature.

File: src/hackerrank-problems/camel-case4.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+function splitCamelCase(input: string): string {
+    // Destructure input string
+    const [, format, name] = input.split(";")
+    // Split input based on uppercase char
+    const words = name
+        .replace(/([a-z])([A-Z])/g, "$1 $2")
+        .toLowerCase()
+        .split(" ")
```
Comment (pancudaniel7) [inline src/hackerrank-problems/camel-case4.ts:46]: 👍🏻

**Review by pancudaniel7 — CHANGES_REQUESTED** [review]

File: src/hackerrank-problems/breaking-the-records.ts (hunk 1 above)
Comment (mathias-onu) [reply src/hackerrank-problems/breaking-the-records.ts]: Oh, ok, I see - with array destructuring, forEach loop and clearn if-statements 👍

File: src/hackerrank-problems/camel-case4.ts (hunk 2 above)
Comment (mathias-onu) [reply src/hackerrank-problems/camel-case4.ts]: Ok, got it - I removed the unnecessary comments.

**Pull request merged**